lightning/src/chain/channelmonitor.rs

```diff
@@ -1542,20 +1550,25 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		}
 		let htlc_resolved = self.htlcs_resolved_on_chain.iter()
 			.find(|v| if v.commitment_tx_output_idx == Some(htlc_commitment_tx_output_idx) {
-				debug_assert!(htlc_spend_txid_opt.is_none());
-				htlc_spend_txid_opt = v.resolving_txid;
+				htlc_spend_txid_opt = v.resolving_txid.as_ref();
+				debug_assert!(htlc_spend_tx_opt.is_none());
+				htlc_spend_tx_opt = v.resolving_tx.as_ref();
 				true
 			} else { false });
 		debug_assert!(holder_timeout_spend_pending.is_some() as u8 + htlc_spend_pending.is_some() as u8 + htlc_resolved.is_some() as u8 <= 1);
 
+		let htlc_commitment_outpoint = BitcoinOutPoint::new(confirmed_txid.unwrap(), htlc_commitment_tx_output_idx);
 		let htlc_output_to_spend =
-			if let Some(txid) = htlc_spend_txid_opt {
-				debug_assert!(
-					self.onchain_tx_handler.channel_transaction_parameters.opt_anchors.is_none(),
-					"This code needs updating for anchors");
-				BitcoinOutPoint::new(txid, 0)
+			if let Some(ref tx) = htlc_spend_tx_opt {
+				// Because HTLCs are signed with SIGHASH_SINGLE, we can locate the correct output
```

Comment:
They're also signed with `SIGHASH_ANYONECANPAY`, I think this can also precised this is the behavior under BIP143 (in case a future algorithm digest modifies this assumption)